app/post_metadata.go

```diff
@@ -383,74 +417,154 @@ func (a *App) getImagesInMessageAttachments(post *model.Post) []string {
 	return images
 }
 
-func (a *App) getLinkMetadata(requestURL string, timestamp int64, isNewPost bool) (*opengraph.OpenGraph, *model.PostImage, error) {
+func looksLikeAPermalink(url, siteURL string) bool {
+	expression := fmt.Sprintf(`^(%s).*(/pl/)[a-z0-9]{26}$`, siteURL)
```

Comment:
We could maybe support relative links, but that may be an edge case for our community/community-daily case. Not sure how common that is. Where are you 0-5 that we should support relative links out of the gate?

What's the sneaky regex thing you're talking about? I didn't follow.

Reply:
I'm 0/5 on whether or not we ship this without support for community, but relative link support is 100% necessary to have this work on community since it uses the relative links to work across servers in the cluster. But that also needs https://mattermost.atlassian.net/browse/MM-29224 anyway because you need to use a Markdown link to do a relative link.

As for the sneaky bit, I was trying to figure out if there was any way to trick the regex and have it return true for something that wasn't a valid permalink, but I couldn't come up with any way to actually abuse that, so it's probably not an issue.

Reply:
> relative link support is 100% necessary to have this work on community

Yes, totally agree. There was some talk about disabling that plugin, but I'll take a look at what's involved in supporting it out-of-the-gate and I'll either add it to this PR and we can review the changes or I'll add it as a separate ticket for afterwards.

Reply:
Recapping what we talked about offline: supporting the relative links used by community servers depends upon first [supporting markdown links](https://mattermost.atlassian.net/browse/MM-29224). When that's complete we can make a quick change to the regex so that relative permalinks embed the referenced post info.